src/components/Navbar/UserOptions.js

```diff
@@ -22,12 +23,18 @@ const RightItem = () => {
   const emailHash = md5(user?.email || '');
   const gravatarURL = `https://www.gravatar.com/avatar/${emailHash}`;
 
+  const ref = useRef();
+
   const handleRedirectTo = (path) => history.push(path);
 
   const toggleNavMenu = () => setNavOpen((prevState) => !prevState);
 
+  useClickAway(ref, () => {
+    setNavOpen(false);
+  });
+
   return (
-    <>
+    <div ref={ref}>
```

Comment:
If you have to add a `div` to solve this logic maybe this shouldn't be here. Away from which div do you really want to capture click?

Reply:
I needed to add a wrapper that contained both `Avatar` and `Menu` in order to achieve this effect. If I don't have this extra div, the next parent would be the whole `NavContent` and that's not what we want.